server/internal/client/client.go

```diff
@@ -228,18 +228,18 @@ func SpawnConn(conn Conn) *ConnHandle {
 // If Destroy is called on a ConnHandle which has already been destroyed, the
 // behavior is undefined.
 func (c *ConnHandle) Destroy() {
-	wasDestroyed := !atomic.CompareAndSwapUint32(&c.destroyed, 0, 1)
-	if wasDestroyed {
-		// This isn't technically necessary - we document that behavior in this case
-		// is undefined - but it's nice and makes it less likely that bugs will slip
-		// by unnoticed.
-		panic("destroy already-destroyed ClientHandle")
-	}
 	c.destroy()
 	c.wg.Wait()
 }
 
 func (c *ConnHandle) destroy() {
+	// This can be called explicitly, b/c the health check has failed, or both.
```

Comment:
This comment broke my English parser temporarily where I read a parenthetical phrase not a list.  Maybe "This function could be called both by the health checker and explicitly" would flow a bit better?

Reply:
haha sure